app/assets/javascripts/beak/widgets/ractives/chooser.coffee

```diff
@@ -154,12 +154,18 @@ RactiveChooser = RactiveValueWidget.extend({
 
   widgetType: "chooser"
 
-  observe: {
-    'widget.currentValue': () ->
-      widget        = @get('widget')
-      currentChoice = widget.choices.findIndex( (c) -> c is widget.currentValue )
-      @set('widget.currentChoice', if currentChoice >= 0 then currentChoice else 0)
-      return
+  on: {
+    'chooser-option-change': (event) ->
+      widget       = @get('widget')
+      selectedIdx  = parseInt(event.node.value)
+      selectedChoice = widget.choices[selectedIdx] or widget.choices[0]
```

Comment:
It won't cause an error, but it seems odd to me to get the `selectedChoice` before checking if the index is valid.  I think it could be moved inside the `if isSelectedIdxValid` conditional?  That might also remove the need to mix in the `or widget.choices[0]`.